<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SerialAssistant</class>
 <widget class="QWidget" name="SerialAssistant">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1090</width>
    <height>678</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SerialAssistant</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QGroupBox" name="QGBCtrlRx">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>230</y>
     <width>171</width>
     <height>101</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QGroupBox{
border-width:1px;
border-style:solid;
border-radius:4px;
font:bold ; 
}</string>
   </property>
   <property name="title">
    <string>通用接收控制</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
   <property name="checkable">
    <bool>false</bool>
   </property>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>73</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QCheckBox" name="chkDisplayHex">
       <property name="text">
        <string>Hex显示</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkBox_5">
       <property name="text">
        <string>加时间戳</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="btnSaveFile">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>49</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>保存至文件...</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="btnClear">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>21</y>
      <width>71</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>清空</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnPause">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>21</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>暂停</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="QGBCtrlTx">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>340</y>
     <width>171</width>
     <height>261</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QGroupBox{
border-width:1px;
border-style:solid;
border-radius:4px;
font:bold ; 
}</string>
   </property>
   <property name="title">
    <string>自动发送控制</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
   <property name="checkable">
    <bool>false</bool>
   </property>
   <widget class="QSpinBox" name="spinBox_2">
    <property name="geometry">
     <rect>
      <x>65</x>
      <y>230</y>
      <width>61</width>
      <height>18</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>61</width>
      <height>18</height>
     </size>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="suffix">
     <string/>
    </property>
    <property name="minimum">
     <number>30</number>
    </property>
    <property name="maximum">
     <number>99999</number>
    </property>
    <property name="value">
     <number>99</number>
    </property>
   </widget>
   <widget class="QLabel" name="lblUnit_ms_2">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>230</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>ms</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblTimedTx">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>230</y>
      <width>48</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>定时发送</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_8">
     <item>
      <widget class="QLabel" name="lblTerminator">
       <property name="text">
        <string>结尾自动添加</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cbbTerminator">
       <item>
        <property name="text">
         <string>NONE</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CR</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>LF</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CR\LF</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>151</width>
      <height>20</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <item>
      <widget class="QCheckBox" name="chbTxHex">
       <property name="text">
        <string>Hex发送</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="chbTimedTx">
       <property name="text">
        <string>定时发送</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="lblFileSize">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>文件大小：</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>151</width>
      <height>80</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="1" column="0">
      <widget class="QLineEdit" name="lineEdit_3"/>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="lblVerifyPos">
       <property name="text">
        <string>校验码插入位置</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLineEdit" name="lineEdit"/>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="lblBeginPos">
       <property name="text">
        <string>起始位置</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="lblByteNum">
       <property name="text">
        <string>字节数</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLineEdit" name="lineEdit_2"/>
     </item>
    </layout>
   </widget>
   <widget class="QComboBox" name="cbbVerifyStyle">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>101</y>
      <width>151</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>添加检验码</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>ADD8</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>0-ADD8</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>XOR8</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>CRC16(MUDBUS)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>固定字节</string>
     </property>
    </item>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>22</y>
      <width>151</width>
      <height>25</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QPushButton" name="btnLoadFile">
       <property name="text">
        <string>加载文件</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnTxFile">
       <property name="text">
        <string>传输文件</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QProgressBar" name="progressBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>646</y>
     <width>171</width>
     <height>16</height>
    </rect>
   </property>
   <property name="cursor">
    <cursorShape>ArrowCursor</cursorShape>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
   <property name="value">
    <number>0</number>
   </property>
   <property name="textVisible">
    <bool>true</bool>
   </property>
   <property name="invertedAppearance">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QToolBox" name="toolBox">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>420</y>
     <width>881</width>
     <height>251</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>1</number>
   </property>
   <widget class="QWidget" name="page_1">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>881</width>
      <height>199</height>
     </rect>
    </property>
    <attribute name="label">
     <string>单条发送</string>
    </attribute>
    <widget class="QPushButton" name="btnSend">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>71</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>发送</string>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="txtSingle">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>0</y>
       <width>761</width>
       <height>201</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="page_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>881</width>
      <height>199</height>
     </rect>
    </property>
    <attribute name="label">
     <string>多条发送</string>
    </attribute>
    <widget class="QWidget" name="layoutWidget7">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>881</width>
       <height>161</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="4" column="4">
       <widget class="QPushButton" name="btnMultiPush10">
        <property name="text">
         <string>10</string>
        </property>
       </widget>
      </item>
      <item row="4" column="5">
       <widget class="QLineEdit" name="txtLine10"/>
      </item>
      <item row="3" column="5">
       <widget class="QLineEdit" name="txtLine09"/>
      </item>
      <item row="3" column="4">
       <widget class="QPushButton" name="btnMultiPush09">
        <property name="text">
         <string>9</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="chkMulti02">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="btnMultiPush02">
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="btnMultiPush06">
        <property name="text">
         <string>6</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLineEdit" name="txtLine03"/>
      </item>
      <item row="0" column="2">
       <widget class="QLineEdit" name="txtLine01"/>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="btnMultiPush03">
        <property name="text">
         <string>3</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLineEdit" name="txtLine02"/>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="btnMultiPush04">
        <property name="text">
         <string>4</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QPushButton" name="btnMultiPush05">
        <property name="text">
         <string>5</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLineEdit" name="txtLine04"/>
      </item>
      <item row="4" column="2">
       <widget class="QLineEdit" name="txtLine05"/>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="btnMultiPush01">
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QLineEdit" name="txtLine06"/>
      </item>
      <item row="2" column="5">
       <widget class="QLineEdit" name="txtLine08"/>
      </item>
      <item row="1" column="4">
       <widget class="QPushButton" name="btnMultiPush07">
        <property name="text">
         <string>7</string>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QLineEdit" name="txtLine07"/>
      </item>
      <item row="2" column="4">
       <widget class="QPushButton" name="btnMultiPush08">
        <property name="text">
         <string>8</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="chkMulti01">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QCheckBox" name="chkMulti05">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="chkMulti03">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="chkMulti04">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QCheckBox" name="chkMulti06">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QCheckBox" name="chkMulti07">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QCheckBox" name="chkMulti08">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QCheckBox" name="chkMulti09">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QCheckBox" name="chkMulti10">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QGroupBox" name="QGBSerialConfig">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>171</width>
     <height>211</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QGroupBox{
border-width:1px;
border-style:solid;
border-radius:4px;
font:bold ; 
}</string>
   </property>
   <property name="title">
    <string>串口设置</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
   <property name="checkable">
    <bool>false</bool>
   </property>
   <widget class="QPushButton" name="btnOpenPort">
    <property name="geometry">
     <rect>
      <x>89</x>
      <y>161</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string>打开串口</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <property name="autoDefault">
     <bool>false</bool>
    </property>
    <property name="default">
     <bool>false</bool>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QWidget" name="formLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>151</width>
      <height>136</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <widget class="QLabel" name="lblPortNum">
       <property name="text">
        <string>端口号</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="cbbPortNum">
       <property name="maxVisibleItems">
        <number>20</number>
       </property>
       <property name="sizeAdjustPolicy">
        <enum>QComboBox::AdjustToContents</enum>
       </property>
       <property name="minimumContentsLength">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="lblBaudRate">
       <property name="text">
        <string>波特率</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="cbbBaud">
       <property name="sizeAdjustPolicy">
        <enum>QComboBox::AdjustToContents</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="lblDataBit">
       <property name="text">
        <string>数据位</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QComboBox" name="cbbDataBit">
       <property name="sizeAdjustPolicy">
        <enum>QComboBox::AdjustToContents</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="lblVerify">
       <property name="text">
        <string>校验</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="lblStopBit">
       <property name="text">
        <string>停止位</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QComboBox" name="cbbStopBit">
       <property name="sizeAdjustPolicy">
        <enum>QComboBox::AdjustToContents</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QComboBox" name="cbbVerify">
       <property name="sizeAdjustPolicy">
        <enum>QComboBox::AdjustToContents</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="btnRefresh">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>161</y>
      <width>72</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>刷新</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="chkFlowCtrlDTR">
    <property name="geometry">
     <rect>
      <x>31</x>
      <y>191</y>
      <width>43</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>DTR</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="chkFlowCtrlRTS">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>191</y>
      <width>43</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>RTS</string>
    </property>
   </widget>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>21</x>
     <y>610</y>
     <width>151</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Rx:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="lblrxCnt">
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Tx:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="lblrxCnt_2">
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPlainTextEdit" name="textBrower">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>10</y>
     <width>881</width>
     <height>401</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>btnClear</sender>
   <signal>clicked()</signal>
   <receiver>textBrower</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>316</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
